storage/storage.go

```diff
@@ -1132,6 +1135,39 @@ func (o *ObjectAttrs) toRawObject(bucket string) *raw.Object {
 	}
 }
 
+// toProtoObject copies the editable attributes from o to the proto library's Object type.
+func (o *ObjectAttrs) toProtoObject(b string) *storagepb.Object {
+	var ret *timestamppb.Timestamp
+	if !o.RetentionExpirationTime.IsZero() {
+		ret = timestamppb.New(o.RetentionExpirationTime)
+	}
+	var ct *timestamppb.Timestamp
+	if !o.CustomTime.IsZero() {
+		ct = timestamppb.New(o.CustomTime)
+	}
+	// For now, there are only globally unique buckets, and "_" is the alias
+	// project ID for such buckets.
+	b = bucket("_", b)
```

Comment:
Missed this in the read PR but we should probably rename this to indicate that it's the proto bucket path.

Reply:
Oop, sure thing.